### DISCUSSION TITLE
How to quickly count the number of files in a filer directory?

### DISCUSSION THREAD

**Discussion by gaozh424** [seaweedfs/seaweedfs#5547] [Q&A]
Hello everyone, I am currently using SeaweedFS to store a large number of small files, and I use the filer for uploading and downloading files. Now, there is a requirement to quickly count the total number of files in each directory under the filer. The only method I have found so far is traversal, but that is too slow. Is there a better way? 
Thank you very much.

**Comment by SmsS4** [comment]
Maybe count from filer store instead of filer?

**Comment by SigmaStar** [comment]
I got an idea
if you use leveldb2 as filer storage and if all your files are named with its md5 value, you can parse leveldb2 database and count how many files are there actually. This is far more fast (several seconds) than using **find** (hours) in fuse filesystem.
Following are my code:

```python
import plyvel # using plyvel for parsing leveldb2
import os
from tqdm import tqdm

def all_ascii(b):
    for i in b:
        if chr(i) not in '0123456789abcdef':
            return False
    return True

res = {}
for d in os.listdir('db'):
    db = plyvel.DB(f'db/{d}')
    for i in tqdm(range(0,256), desc = d):
        for j in db.prefixed_db(int.to_bytes(i,1,'little')):
            if all_ascii(j[0][-32:]): # since all my files are named with md5 value
                res.update({j[0][-32:]:None})
print(f'file counts: {len(res)}')
```

if you name your files with other format, then you need to parse the key structure of seaweedfs. If you got script please share.
